userspace/libscap/engine/bpf/scap_bpf.c

```diff
@@ -1721,18 +1721,20 @@ const struct scap_stats_v2* scap_bpf_get_stats_v2(struct scap_engine_handle engi
 	// we can't collect libbpf stats if bpf stats are not enabled
 	if (!(handle->m_flags & ENGINE_FLAG_BPF_STATS_ENABLED))
 	{
-		flags &= ~PPM_SCAP_STATS_LIBBPF_STATS;
+		flags &= ~METRICS_V2_LIBBPF_STATS;
 	}
 
-	if ((flags & PPM_SCAP_STATS_KERNEL_COUNTERS) && (BPF_MAX_KERNEL_COUNTERS_STATS <= nstats_allocated))
+	if ((flags & METRICS_V2_KERNEL_COUNTERS) && (BPF_MAX_KERNEL_COUNTERS_STATS <= nstats_allocated))
 	{
 		/* KERNEL SIDE STATS COUNTERS */
 		for(int stat = 0; stat < BPF_MAX_KERNEL_COUNTERS_STATS; stat++)
 		{
-			stats[stat].type = STATS_VALUE_TYPE_U64;
-			stats[stat].flags = PPM_SCAP_STATS_KERNEL_COUNTERS;
+			stats[stat].type = METRIC_VALUE_TYPE_U64;
```

Comment:
I start to think that an helper method to fill stats would be a good addition (perhaps even in a followup PR), like:
```
fill_stats(&stats[stat], type, value, unit, metric_type, flags);
```

Reply:
Tracking it here https://github.com/falcosecurity/libs/issues/1463#issuecomment-1992835678 

The `new_metric` method is a template in CPP.  In addition for the scap stats for example we have inner loops where we only update the value, e.g. see https://github.com/falcosecurity/libs/blob/5dc692e3045f5102e1f7752d40ec7e4f81cbc29e/userspace/libscap/engine/bpf/scap_bpf.c#L1738 when we iterate over the CPUs. Therefore a bit on the fence as I see the value of it being nicer from a software organization point of view, but the current approach seems faster. 

Let's think more about it? I believe the real dilemma of `metrics` is that one part is in C while the other part is in CPP, so it's difficult to find a shared approach that would follow best practices either way. I'll repeat this in my next response below. That's also why I use the arrays approach with the metrics names as that's how we do it in scap / C. I am still not sure, how a really great metrics framework that spans scap and sinsp + pending falco rules counts metrics category should look like frankly.

Reply:
>  I believe the real dilemma of metrics is that one part is in C while the other part is in CPP

Yep, you are right, i though about this thing later on. I am not sure how (and if we can) to proceed, i agree to "save the idea" for later; let's see if anyone comes with a good solution.